### ISSUE TITLE
Command log mode does not start correctly

### ISSUE THREAD

**Issue by aeberts** [syl20bnr/spacemacs#7743]
#### Description :octocat:

Command log mode does not start correctly

#### Reproduction guide :beetle:
- Start Emacs
- enable command log mode in user init
- restart emacs
- from normal mode type: ```SPC a L```

*Observed behaviour:* :eyes: :broken_heart:

-> new emacs buffers are created and spacemacs becomes unusable (see screenshot)

*Expected behaviour:* :heart: :smile:

-> command mode is enabled. 

#### System Info :computer:
- OS: darwin
- Emacs: 25.1.1
- Spacemacs: 0.200.5
- Spacemacs branch: master (rev. 664ba6a)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(osx
 (auto-completion :variables auto-completion-enable-snippets-in-popup t auto-completion-enable-help-tooltip t auto-completion-enable-sort-by-usage t)
 better-defaults emacs-lisp
 (org :variables org-enable-github-support t)
 git github markdown clojure python html erc
 (shell :variables shell-default-height 30 shell-default-position 'bottom)
 spell-checking syntax-checking version-control themes-megapack command-log)
```
- System configuration features: JPEG RSVG IMAGEMAGICK NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS

#### Backtrace :paw_prints:
```
<<BACKTRACE IF RELEVANT>>
```
![screen shot 2016-11-15 at 9 29 51 am](https://cloud.githubusercontent.com/assets/571576/20316528/126cfa4a-ab17-11e6-93bd-40f4ccd5794f.png)

**Comment by TheBB** [comment]
I can't reproduce this. Please try from a base installation and provide better reproduction steps.

**Comment by aeberts** [comment]
Just tried with a completely clean install.

```SPC a L``` appears to work fine in an *.el or *.clj buffer.

Error occurs when enabling command-log from the ```*spacemacs*``` buffer with ```SPC a L```

## Backtrace

```
Debugger entered--Lisp error: (wrong-type-argument wholenump -6)
  make-string(-6 32)
  (insert (make-string (floor (/ (- maxcol badge-size) 2)) 32))
  (progn (insert (make-string (floor (/ (- maxcol badge-size) 2)) 32)) (insert-image badge))
  (if badge (progn (insert (make-string (floor (/ (- maxcol badge-size) 2)) 32)) (insert-image badge)))
  (progn (goto-char (point-max)) (spacemacs-buffer/insert-page-break) (insert "\n") (if badge (progn (insert (make-string (floor (/ (- maxcol badge-size) 2)) 32)) (insert-image badge))) (if heart (progn (if badge (progn (insert "\n\n"))) (insert (make-string (floor (/ (- maxcol ... heart-size ...) 2)) 32)) (insert build-lhs) (insert-image heart) (insert build-rhs) (insert "\n"))))
  (if (or badge heart) (progn (goto-char (point-max)) (spacemacs-buffer/insert-page-break) (insert "\n") (if badge (progn (insert (make-string (floor (/ ... 2)) 32)) (insert-image badge))) (if heart (progn (if badge (progn (insert "\n\n"))) (insert (make-string (floor (/ ... 2)) 32)) (insert build-lhs) (insert-image heart) (insert build-rhs) (insert "\n")))))
  (let* ((maxcol spacemacs-buffer--banner-length) (badge-path spacemacs-badge-official-png) (badge (if (and (display-graphic-p) (image-type-available-p (intern (file-name-extension badge-path)))) (progn (create-image badge-path)))) (badge-size (if badge (progn (car (image-size badge))))) (heart-path spacemacs-purple-heart-png) (heart (if (and (display-graphic-p) (image-type-available-p (intern (file-name-extension badge-path)))) (progn (create-image heart-path)))) (heart-size (if heart (progn (car (image-size heart))))) (build-lhs "Made with ") (build-rhs " by the community") (buffer-read-only nil)) (if (or badge heart) (progn (goto-char (point-max)) (spacemacs-buffer/insert-page-break) (insert "\n") (if badge (progn (insert (make-string (floor ...) 32)) (insert-image badge))) (if heart (progn (if badge (progn (insert "\n\n"))) (insert (make-string (floor ...) 32)) (insert build-lhs) (insert-image heart) (insert build-rhs) (insert "\n"))))))
  (save-excursion (let* ((maxcol spacemacs-buffer--banner-length) (badge-path spacemacs-badge-official-png) (badge (if (and (display-graphic-p) (image-type-available-p (intern ...))) (progn (create-image badge-path)))) (badge-size (if badge (progn (car (image-size badge))))) (heart-path spacemacs-purple-heart-png) (heart (if (and (display-graphic-p) (image-type-available-p (intern ...))) (progn (create-image heart-path)))) (heart-size (if heart (progn (car (image-size heart))))) (build-lhs "Made with ") (build-rhs " by the community") (buffer-read-only nil)) (if (or badge heart) (progn (goto-char (point-max)) (spacemacs-buffer/insert-page-break) (insert "\n") (if badge (progn (insert (make-string ... 32)) (insert-image badge))) (if heart (progn (if badge (progn ...)) (insert (make-string ... 32)) (insert build-lhs) (insert-image heart) (insert build-rhs) (insert "\n")))))))
  spacemacs-buffer//insert-footer()
  (progn (configuration-layer/display-summary emacs-start-time) (if dotspacemacs-startup-lists (progn (spacemacs-buffer/insert-startupify-lists))) (spacemacs-buffer//insert-footer) (spacemacs-buffer/set-mode-line spacemacs--default-mode-line) (force-mode-line-update) (spacemacs-buffer-mode))
  (if (and (boundp (quote spacemacs-initialized)) spacemacs-initialized) (progn (configuration-layer/display-summary emacs-start-time) (if dotspacemacs-startup-lists (progn (spacemacs-buffer/insert-startupify-lists))) (spacemacs-buffer//insert-footer) (spacemacs-buffer/set-mode-line spacemacs--default-mode-line) (force-mode-line-update) (spacemacs-buffer-mode)) (add-hook (quote emacs-startup-hook) (quote spacemacs-buffer//startup-hook) t))
  (save-excursion (if (> (buffer-size) 0) (progn (set (quote save-line) (line-number-at-pos)) (let ((inhibit-read-only t)) (erase-buffer)))) (spacemacs-buffer/set-mode-line "") (setq spacemacs-buffer--note-widgets nil) (spacemacs-buffer/insert-banner-and-buttons) (if (and (boundp (quote spacemacs-initialized)) spacemacs-initialized) (progn (configuration-layer/display-summary emacs-start-time) (if dotspacemacs-startup-lists (progn (spacemacs-buffer/insert-startupify-lists))) (spacemacs-buffer//insert-footer) (spacemacs-buffer/set-mode-line spacemacs--default-mode-line) (force-mode-line-update) (spacemacs-buffer-mode)) (add-hook (quote emacs-startup-hook) (quote spacemacs-buffer//startup-hook) t)))
  (save-current-buffer (set-buffer (get-buffer-create spacemacs-buffer-name)) (page-break-lines-mode) (save-excursion (if (> (buffer-size) 0) (progn (set (quote save-line) (line-number-at-pos)) (let ((inhibit-read-only t)) (erase-buffer)))) (spacemacs-buffer/set-mode-line "") (setq spacemacs-buffer--note-widgets nil) (spacemacs-buffer/insert-banner-and-buttons) (if (and (boundp (quote spacemacs-initialized)) spacemacs-initialized) (progn (configuration-layer/display-summary emacs-start-time) (if dotspacemacs-startup-lists (progn (spacemacs-buffer/insert-startupify-lists))) (spacemacs-buffer//insert-footer) (spacemacs-buffer/set-mode-line spacemacs--default-mode-line) (force-mode-line-update) (spacemacs-buffer-mode)) (add-hook (quote emacs-startup-hook) (quote spacemacs-buffer//startup-hook) t))))
  (progn (setq spacemacs-buffer--banner-length (window-width) spacemacs-buffer--last-width spacemacs-buffer--banner-length) (save-current-buffer (set-buffer (get-buffer-create spacemacs-buffer-name)) (page-break-lines-mode) (save-excursion (if (> (buffer-size) 0) (progn (set (quote save-line) (line-number-at-pos)) (let ((inhibit-read-only t)) (erase-buffer)))) (spacemacs-buffer/set-mode-line "") (setq spacemacs-buffer--note-widgets nil) (spacemacs-buffer/insert-banner-and-buttons) (if (and (boundp (quote spacemacs-initialized)) spacemacs-initialized) (progn (configuration-layer/display-summary emacs-start-time) (if dotspacemacs-startup-lists (progn (spacemacs-buffer/insert-startupify-lists))) (spacemacs-buffer//insert-footer) (spacemacs-buffer/set-mode-line spacemacs--default-mode-line) (force-mode-line-update) (spacemacs-buffer-mode)) (add-hook (quote emacs-startup-hook) (quote spacemacs-buffer//startup-hook) t)))) (if save-line (progn (goto-char (point-min)) (forward-line (1- save-line)) (forward-to-indentation 0)) (spacemacs-buffer/goto-link-line)) (switch-to-buffer spacemacs-buffer-name) (spacemacs//redisplay))
  (if (or (not (eq spacemacs-buffer--last-width (window-width))) (not buffer-exists) refresh) (progn (setq spacemacs-buffer--banner-length (window-width) spacemacs-buffer--last-width spacemacs-buffer--banner-length) (save-current-buffer (set-buffer (get-buffer-create spacemacs-buffer-name)) (page-break-lines-mode) (save-excursion (if (> (buffer-size) 0) (progn (set (quote save-line) (line-number-at-pos)) (let (...) (erase-buffer)))) (spacemacs-buffer/set-mode-line "") (setq spacemacs-buffer--note-widgets nil) (spacemacs-buffer/insert-banner-and-buttons) (if (and (boundp (quote spacemacs-initialized)) spacemacs-initialized) (progn (configuration-layer/display-summary emacs-start-time) (if dotspacemacs-startup-lists (progn ...)) (spacemacs-buffer//insert-footer) (spacemacs-buffer/set-mode-line spacemacs--default-mode-line) (force-mode-line-update) (spacemacs-buffer-mode)) (add-hook (quote emacs-startup-hook) (quote spacemacs-buffer//startup-hook) t)))) (if save-line (progn (goto-char (point-min)) (forward-line (1- save-line)) (forward-to-indentation 0)) (spacemacs-buffer/goto-link-line)) (switch-to-buffer spacemacs-buffer-name) (spacemacs//redisplay)))
  (let ((buffer-exists (buffer-live-p (get-buffer spacemacs-buffer-name))) (save-line nil)) (if (or (not (eq spacemacs-buffer--last-width (window-width))) (not buffer-exists) refresh) (progn (setq spacemacs-buffer--banner-length (window-width) spacemacs-buffer--last-width spacemacs-buffer--banner-length) (save-current-buffer (set-buffer (get-buffer-create spacemacs-buffer-name)) (page-break-lines-mode) (save-excursion (if (> (buffer-size) 0) (progn (set ... ...) (let ... ...))) (spacemacs-buffer/set-mode-line "") (setq spacemacs-buffer--note-widgets nil) (spacemacs-buffer/insert-banner-and-buttons) (if (and (boundp ...) spacemacs-initialized) (progn (configuration-layer/display-summary emacs-start-time) (if dotspacemacs-startup-lists ...) (spacemacs-buffer//insert-footer) (spacemacs-buffer/set-mode-line spacemacs--default-mode-line) (force-mode-line-update) (spacemacs-buffer-mode)) (add-hook (quote emacs-startup-hook) (quote spacemacs-buffer//startup-hook) t)))) (if save-line (progn (goto-char (point-min)) (forward-line (1- save-line)) (forward-to-indentation 0)) (spacemacs-buffer/goto-link-line)) (switch-to-buffer spacemacs-buffer-name) (spacemacs//redisplay))))
  spacemacs-buffer/goto-buffer()
  (progn (select-window (car save-selected-window--state) (quote norecord)) (spacemacs-buffer/goto-buffer))
  (unwind-protect (progn (select-window (car save-selected-window--state) (quote norecord)) (spacemacs-buffer/goto-buffer)) (internal--after-with-selected-window save-selected-window--state))
  (save-current-buffer (unwind-protect (progn (select-window (car save-selected-window--state) (quote norecord)) (spacemacs-buffer/goto-buffer)) (internal--after-with-selected-window save-selected-window--state)))
  (let ((save-selected-window--state (internal--before-with-selected-window space-win))) (save-current-buffer (unwind-protect (progn (select-window (car save-selected-window--state) (quote norecord)) (spacemacs-buffer/goto-buffer)) (internal--after-with-selected-window save-selected-window--state))))
  (progn (let ((save-selected-window--state (internal--before-with-selected-window space-win))) (save-current-buffer (unwind-protect (progn (select-window (car save-selected-window--state) (quote norecord)) (spacemacs-buffer/goto-buffer)) (internal--after-with-selected-window save-selected-window--state)))))
  (if (and dotspacemacs-startup-buffer-responsive space-win (not (window-minibuffer-p frame-win))) (progn (let ((save-selected-window--state (internal--before-with-selected-window space-win))) (save-current-buffer (unwind-protect (progn (select-window (car save-selected-window--state) (quote norecord)) (spacemacs-buffer/goto-buffer)) (internal--after-with-selected-window save-selected-window--state))))))
  (let ((space-win (get-buffer-window spacemacs-buffer-name)) (frame-win (frame-selected-window))) (if (and dotspacemacs-startup-buffer-responsive space-win (not (window-minibuffer-p frame-win))) (progn (let ((save-selected-window--state (internal--before-with-selected-window space-win))) (save-current-buffer (unwind-protect (progn (select-window ... ...) (spacemacs-buffer/goto-buffer)) (internal--after-with-selected-window save-selected-window--state)))))))
  spacemacs-buffer//resize-on-hook()
```

**Comment by bmag** [comment]
I can reproduce this behavior if I set `dotspacemacs-startup-buffer-responsive` to `t`. The backtrace shows an error in the code that resizes the home buffer. Until this is fixed, I suggest setting `dotspacemacs-startup-buffer-responsive` to `nil` as a work-around.

**Comment by TheBB** [comment]
@deb0ch Something tells me you're the right man to ping.

I still can't reproduce it though.

**Comment by TinySong** [comment]
Spacemacs develop branch has the same error

**Comment by deb0ch** [comment]
Ok, I can reproduce :mask: 

Reproduction steps I took:

- activate `command-log` layer
- in `*spacemacs*` buffer, create tons of new windows with `SPC w v`
- when trying to use `SPC b b` I get the error `error: (wrong-type-argument wholenump -6)`
- whenever `SPC b b` displays, it does in unusual / inappropriate places

Tried again without `command-log` and same behaviour, this particular error seems unrelated to this layer.

However, there is another error related to `command-log-mode`, and this one is very funny indeed :laughing: 

- start Spacemacs
- in the home buffer, only one window is active: the home buffer window
- `SPC a L`
- creates by itself a :shit: ton of new windows until there is the above-mentioned `error: (wrong-type-argument wholenump -6)`, which seems to be stopping this crazy process :scream_cat:   
  
Will start investigating whenever I can :cop: :construction_worker: 

@TheBB I wonder why :joy_cat:

**Comment by deb0ch** [comment]
#7874 should partially fix the issue, at least the part where the
`error: (wrong-type-argument wholenump -6)` is thrown.

Though I doubt that it will fix the crazy window creation problem, keep me updated of the results :smiley_cat:

**Comment by d12frosted** [comment]
That PR is in `develop`. @aeberts could you please check it? :)

**Comment by aeberts** [comment]
Would be happy to check the PR but I'll need a little help as I'm not seeing the commit id for #7874 on develop (I'm a git newbie so please bear with me). Here's what I'm doing:

* I installed Spacemacs stable using instructions on the main page.
* I do `git checkout develop` in my `~\.emacs.d\`
* `git fetch`
* `git show <commit id>` => `fatal: bad object <commit id>`
* `git log` also doesn't show the commit id.

Any idea what I'm doing wrong here?

**Comment by deb0ch** [comment]
Look for the commit message in the `git log`.

Commit SHA1s depend not only on the changes, but also on the previous commits in the commit history.

So when @d12frosted cherry-picked this commit on `develop` he added it at a different place in the commit history, which could happen for instance if I didn't update my local branch for some time.

Is it any clearer now ?

**Comment by aeberts** [comment]
@deb0ch  Yes, thanks. That totally makes sense (I appreciate the guidance!). 

I tested on my machine and the issue addressed by #7874 appears to be fixed.

Detailed test steps:

* Restart emacs
* activate command-log layer
* in *spacemacs* buffer, create tons of new windows with SPC w v
* use `SPC b b` => helm buffer appears normally.

**Comment by deb0ch** [comment]
Though I would like to point out that the bug addressed in #7874 shouldn't be the root cause of this one. Just that the bug described here was triggering the one that I corrected.
